libraries/Microsoft.Bot.Builder.Expressions/CommonRegex.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+using Antlr4.Runtime;
+using Antlr4.Runtime.Misc;
+using Antlr4.Runtime.Tree;
+
+namespace Microsoft.Bot.Builder.Expressions
+{
+    public class CommonRegex
+    {
+        public static Regex CreateRegex(string pattern)
+        {
+            if (string.IsNullOrEmpty(pattern) || !IsCommanRegex(pattern))
+            {
+                throw new ArgumentException("A regular expression parsing error occurred.");
+            }
+
+            // TODO check pattern
+            return new Regex(pattern, RegexOptions.Compiled);
+        }
+
+        private static bool IsCommanRegex(string pattern)
+        {
+            try
+            {
+                AntlrParse(pattern);
```

Comment:
Given the most common case is a static regex, might be worth moving this to the static analysis if possible so you don't have to do it at runtime unless the pattern is dynamic.

Reply:
yes, agree

Reply:
How can we judge the regex is static or dynamic? by checking if there is a special character in there?

Reply:
If the pattern is a constant regex string then you know it is static.

Reply:
Maybe my poor English is affecting here, I don't quite get what a constant regex string means? Can you give an example of how it looks like, and how we can do a quick check to judge if it's constant regex or not?

Reply:
Fixed